vulnfeeds/cmd/combine-to-osv/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/google/osv/vulnfeeds/cves"
+	"github.com/google/osv/vulnfeeds/vulns"
+)
+
+const (
+	defaultCvePath        = "cve_jsons"
+	defaultPartsInputPath = "parts"
+	defaultOsvOutputPath  = "osv_output"
+)
+
+func main() {
+	cvePath := flag.String("cvePath", defaultCvePath, "Path to CVE file")
+	partsInputPath := flag.String("partsPath", defaultPartsInputPath, "Path to CVE file")
+	osvOutputPath := flag.String("osvOutputPath", defaultOsvOutputPath, "Path to CVE file")
+	flag.Parse()
+
+	err := os.MkdirAll(*cvePath, 0755)
+	if err != nil {
+		log.Fatalf("Can't create output path: %s", err)
+	}
+	err = os.MkdirAll(*osvOutputPath, 0755)
+	if err != nil {
+		log.Fatalf("Can't create output path: %s", err)
+	}
+
+	allCves := loadAllCVEs(*cvePath)
+	allParts := loadParts(*partsInputPath)
+	combinedData := combineIntoOSV(allCves, allParts)
+	writeOsvFile(combinedData, *osvOutputPath)
+}
+
+// loadParts loads files generated by other executables in the cmd folder
+func loadParts(partsInputPath string) map[string][]vulns.PackageInfo {
+	dir, err := os.ReadDir(partsInputPath)
+	if err != nil {
+		log.Fatalf("Failed to read dir? %s", err)
+	}
+	output := map[string][]vulns.PackageInfo{}
+	for _, entry := range dir {
+		if !entry.IsDir() {
+			log.Println("Unexpected file entry in " + partsInputPath)
+			continue
+		}
+		dirInner, err := os.ReadDir(partsInputPath + "/" + entry.Name())
+		if err != nil {
+			log.Fatalf("Failed to read dir? %s", err)
+		}
+
+		for _, entryInner := range dirInner {
+			file, err := os.Open(partsInputPath + "/" + entry.Name() + "/" + entryInner.Name())
+			if err != nil {
+				log.Fatalf("Failed to open cve json: %s", err)
+			}
+			var pkgInfos []vulns.PackageInfo
+			err = json.NewDecoder(file).Decode(&pkgInfos)
+			if err != nil {
+				log.Fatalf("Failed to decode json: %s", err)
+			}
+
+			cveId := strings.Split(entryInner.Name(), ".")[0]
```

Comment:
Can you use https://pkg.go.dev/path#Base instead?

Reply:
Turns out path.base doesn't actually give the file name without extension, it's just for getting the file name in a long path.